### ISSUE TITLE
Two class Segmentation, but loss does not going down

### ISSUE THREAD

**Issue by Parker-Lyu** [CoinCheung/BiSeNet#12]
Hi, 
Thank you for your  sharing.
I changed your network to do a two-class segmentation.But met some probles
what I changed is :
    dataloader: background as 0, target as 1;
    lr_start = 2.5e-6; warmup_start_lr = 1e-7;
    I have 7000 images to train;
    set batch_size=6, epochs=20;
    use resnet18 as backbone;
    train on a single GPU;

What bother me is the  loss:
    loss down to 1.7 from 2.3 in the first 2.5k step, but  after that , loss begin to shake with great amplitude and going up. I adjust learning rate but that does not work, could you give me some advise?

**Comment by CoinCheung** [comment]
Did you train from scratch or from my pretrained models ? What is the cropsize you set? Maybe a batch size of 6 is not enough, it is better to use at least 8. By the way, have you tried to prolong the warmup iterations? Maybe it will help.

**Comment by CoinCheung** [comment]
If you are not convenient to use a large batch size, but you do not mind to spend longer time to train your model. You can consider Group Norm. It also works to ease your optimization, and it does not require you to use large batch size.

**Comment by Parker-Lyu** [comment]
Thank you for your excelent work and your reply.
I'm new to segmentation, but I understand your code and get a good result now.
Thank you very much

**Comment by HITjialanyu** [comment]
@Parker-Lyu  So,do you have some advise in a imbalanced two-class  segmentation?Ratio of the number of positive samples to the number of negative samples is 3:97,so,i changed nn.CrossEntropyLoss weight=[0.97,0.03],the result is that it predict all point as positive samples. Thanks..

**Comment by HITjialanyu** [comment]
@CoinCheung How to train from your pretrained models? since your cpu model, when using  net.load_state_dict(torch.load('model_final.pth',lambda storage, loc: storage.cuda()))  to continue distributed training,it erros...thank you

**Comment by Parker-Lyu** [comment]
> @Parker-Lyu So,do you have some advise in a imbalanced two-class segmentation?Ratio of the number of positive samples to the number of negative samples is 3:97,so,i changed nn.CrossEntropyLoss weight=[0.97,0.03],the result is that it predict all point as positive samples. Thanks..

I did not calculate Ratio of the number of positive samples to the number of negative samples.
I used 'OhemCELoss' from loss.py, set  n_min=all_pixels_nums/2.

**Comment by HITjialanyu** [comment]
THanks!

**Comment by zhangyunming** [comment]
maybe your lr_start is too low, i think.
